skythought/skythought_evals/batch/workload.py

```diff
@@ -0,0 +1,293 @@
+        elif self.dataset_fraction > 1.0:
+            n_rows = int(self.dataset_fraction)
+            logger.info("Taking the first %d rows from dataset", n_rows)
+            ds = ds.limit(n_rows)
+
+        if repartition_by_batch_size:
+            if enable_ckpt:
+                raise ValueError(
+                    "Cannot repartition the dataset by batch size when "
+                    "checkpointing is enabled."
+                )
+            num_requests = ds.count()
+            num_blocks = math.ceil(num_requests / max_batch_size)
+            ds = ds.repartition(num_blocks)
+
+            logger.info("#Requests: %d (%d blocks)", num_requests, num_blocks)
+        else:
+            # When checkpointing is enabled, the number of blocks is unknown
+            # at this point.
+            num_blocks = None
+
+        mapper_fn = (
+            self.parse_row_with_carryover_input
+            if self.carryover_inputs
+            else self.parse_row
+        )
+        return ds.map(mapper_fn), num_blocks
+
+    def parse_row_with_carryover_input(self, row: dict[str, Any]) -> dict[str, Any]:
+        """Same as parse_row but carries over the input keys that are not in the output row.
+
+        This is useful when we want to keep the input keys in the output.
+        This method assumes if user returns the same output keys as
+        input keys they have already copied input over and there is
+        no need to do it again for those keys. We will just copy the input_keys that
+        are not in the output row.
+
+        Args:
+            row: The row to be parsed.
+
+        Returns:
+            The parsed row.
+        """
+        input_row_keys = set(row.keys())
+        output_row = self.parse_row(row)
+        output_row_keys = set(output_row.keys())
+        return {
+            **{k: row[k] for k in input_row_keys if k not in output_row_keys},
+            **output_row,
+        }
+
+    def tokenizer_constructor_kwargs(
+        self, model: str
+    ):
+        """Return the keyword arguments for tokenizer constructor.
+
+        Args:
+            model: The model name.
+
+        Returns:
+            The keyword arguments for tokenizer constructor.
+        """
+        return {"model": model}
+
+    def parse_row(self, row: dict[str, Any]) -> dict[str, Any]:
+        """Parse a row in the dataset.
+
+        Args:
+            row: The row in the dataset.
+
+        Returns:
+            The parsed row.
+        """
+        return row
+
+    def postproc_after_tokenize(self, ds: Dataset) -> Dataset:
+        """Post-process the dataset after tokenization.
+
+        Args:
+            ds: The tokenized dataset.
+
+        Returns:
+            The post-processed dataset.
+        """
+        return ds
+
+@dataclass
+class ChatWorkloadBase(WorkloadBase):
+    """The base class for a chat workload."""
+
+    tokenizer_cls: Any = ChatTemplateTokenizer
+
+@dataclass
+class EvalWorkload(ChatWorkloadBase):
```

Comment:
can you simply merge all of these EvalWorkload ChatWorkload and WorkloadBase classes into one minimal version? 

basically what you need from `inference_and_check` is for `EvalWorkload(dataset=ds, sampling_params={"n": args.n, "max_tokens": max_tokens, "temperature": temp})` to work?

Basically get rid of anything that seems extra from here.